### DISCUSSION TITLE
[INET] Fragmentation / defragmentation blocks

### DISCUSSION THREAD

**Discussion by WolframStrauss** [omnetpp/omnetpp#1025] [Q&A] [answered]
Hi all,
I would like to understand the [fragmentation / defragmentation](https://doc.omnetpp.org/inet/api-current/neddoc/packages.html#inet_protocolelement_fragmentation) process (length based). My starting point is: `inet/tutorials/protocol/Network90.ned` which is not running (the version of the respective [GitHub tutorial branch](https://github.com/inet-framework/inet/tree/topic/protocoltutorial/tutorials/protocol) either).

I tried to strip it down to: `Source -> fragmentation -> defragmentation -> sink`. So far without a working system. I did not find out, how to specify the fragmentation policy in a fragmentation block.
  
Any hint is welcome. Thank you Wolfram

**Comment by levy** [comment, marked as answer]
The stripped down version should work.

```
network TestNetwork
{
    submodules:
        source: ActivePacketSource;
        fragmenter: FragmentNumberHeaderBasedFragmenter;
        defragmenter: FragmentNumberHeaderBasedDefragmenter;
        sink: PassivePacketSink;
    connections:
        source.out --> fragmenter.in;
        fragmenter.out --> defragmenter.in;
        defragmenter.out --> sink.in;
}
```

```
[General]
network = TestNetwork

**.source.packetLength = 1000B
**.source.productionInterval = 1s
**.fragmenter.fragmenterPolicy.typename = "LengthBasedFragmenterPolicy"
**.fragmenter.maxFragmentLength = 100B
```

**Comment by WolframStrauss** [comment]
Hi Levy,
thank you very much for your help. It works with a little addition to the last `ini` entry:
```Code
...
**.fragmenter.fragmenterPolicy.maxFragmentLength = 100B
```
